packages/react-ui/src/components/theme-provider/index.js

```diff
@@ -39,15 +41,38 @@ function ThemeProvider({
     components.Global
   )
 
+  // Extract the styles that need that to be scoped to the
+  // local ThemeProvider
+  const {
+    '*': universalStyles,
+    body,
+    ':root': rootStyles,
+    ...scopedStyles
+  } = components.Global
+
   const combinedTheme = merge(tokens, { components })
 
   // pass a hint to element
   combinedTheme.defined = true
 
+  /**
+   *  Apply global styles only if it's the first ThemeProvider.
+   *  Subsequnt ThemeProviders should have scoped: true and will
+   *  get rest of the styled scoped to the root
+   */
+
   return (
     <Provider theme={combinedTheme} {...props}>
-      <Global styles={interpolate(components.Global, combinedTheme)} />
-      {props.children}
+      {!scoped ? (
+        <>
+          <Global styles={interpolate(components.Global, combinedTheme)} />
+          {props.children}
+        </>
+      ) : (
+        <Element as="div" css={{ scopedStyles }}>
```

Comment:
If the themeprovider is scoped? Looks like it will simply skip tags like body and root. Is that the intention here?

Reply:
Oops! That should be `<Element as="div" css={scopedStyles}>` and it should include the `*` styles.

The ideas was if the styles are really meant for the body or document tags, should they be also passed to the root element? Passing them as such wouldn't make a difference anyway. (https://codesandbox.io/s/modest-sutherland-9wke8)

Reply:
This is where we need to take the call on what really separates the global and scoped styles and how the API for that should look like.

Reply:
> The ideas was if the styles are really meant for the body or document tags, should they be also passed to the root element?

Yeah, should ignore them + throw a warning on console to educate

Reply:
Cool beans! I'll add some warnings + docs to educate users.

One last thing, how do you want the API for styles scoped to the root to look like? Right now it's taking whatever's in `components.Global` minus `body` and `:root`.

e.g it might be a common usecase to expect the body and root to have the same styles. The API to achieve that right now looks like 

```
Global = {
  body:{
    color: 'text.body'
  },
  color: 'text.body'
}
```

It should be a little more explicit no?

Reply:
Some of the options in my mind are

1. Merge everything that's in body with scopedStyles.
2. Add a new `root` key in the components token maybe (or maybe inside components.Global)
3. Pass a `style` prop to ThemeProvider which will be merged with scopedStyles.

Reply:
Hey @siddharthkp 
Have you had the chance to think about this?

I have added warning based on the current API.